bson/src/test/unit/org/bson/GenericBsonTest.java

```diff
@@ -278,26 +278,31 @@ private void runParseError() {
 
 
 
-    // TODO: Working around the fact that the Java driver doesn't report an error for invalid UTF-8, but rather replaces the invalid
+    // Working around the fact that the Java driver doesn't report an error for invalid UTF-8, but rather replaces the invalid
     // sequence with the replacement character
     private void throwIfValueIsStringContainingReplacementCharacter(final String description) {
         BsonDocument decodedDocument = decodeToDocument(testCase.getString("bson").getValue(), description);
-        String testKey = decodedDocument.keySet().iterator().next();
-
-        if (decodedDocument.containsKey(testKey)) {
-            String decodedString = null;
-            if (decodedDocument.get(testKey).isString()) {
-                decodedString = decodedDocument.getString(testKey).getValue();
-            }
-            if (decodedDocument.get(testKey).isDBPointer()) {
-                decodedString = decodedDocument.get(testKey).asDBPointer().getNamespace();
-            }
-            if (decodedString != null && decodedString.contains(Charset.forName("UTF-8").newDecoder().replacement())) {
-                throw new BsonSerializationException("String contains replacement character");
-            }
+        BsonValue value = decodedDocument.get(decodedDocument.getFirstKey());
```

Comment:
Along with adding support for three more BSON types, I simplified the code a bit.

Reply:
Without this change, the Invalid UTF-8 tests don't pass.